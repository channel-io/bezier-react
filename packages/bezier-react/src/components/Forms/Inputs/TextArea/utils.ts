import { type SemanticNames } from '~/src/foundation'

interface CalculateBgColorProps {
  focused: boolean
  hasError: boolean
  readOnly: boolean
}

export function getTextAreaBgColorSemanticName({
  focused,
  hasError,
  readOnly,
}: CalculateBgColorProps): SemanticNames {
  if (readOnly) { return 'bg-grey-lighter' }
  if (focused || hasError) { return 'bg-white-normal' }
  return 'bg-grey-lightest'
}
