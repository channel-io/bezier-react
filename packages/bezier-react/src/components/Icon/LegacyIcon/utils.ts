/* Internal Dependencies */
// eslint-disable-next-line no-restricted-imports
import icons, { type IconName } from '../generated'

/**
 * @deprecated Please import and use individual icons.
 * @example
 * import { Icon, AllIcon, type IconProps } from '@channel.io/bezier-react'
 * <Icon source={AllIcon} color="bg-black-dark" />
 * @example <caption>How to validate the bezier icon source</caption>
 * import { isBezierIcon, AllIcon } from '@channel.io/bezier-react'
 * isBezierIcon(AllIcon) // true
 * isBezierIcon(<svg>...</svg>) // false
 * @example <caption>Legacy icon component is still available. but it will be removed in future versions!</caption>
 * import { LegacyIcon, type LegacyIconProps } from '@channel.io/bezier-react'
 * <LegacyIcon name="all" color="bg-black-dark" />
 */
export const isIconName = (name: any): name is IconName => Object.keys(icons).includes(name)
