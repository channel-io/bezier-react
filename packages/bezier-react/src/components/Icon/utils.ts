import { type SVGProps } from 'react'

const BEZIER_ICON_SYMBOL = Symbol('bezier.icon')

export type IconSource = (props: SVGProps<SVGSVGElement>) => JSX.Element

export type BezierIcon = IconSource & { [BEZIER_ICON_SYMBOL]: true }

export function isBezierIcon(arg: unknown): arg is BezierIcon {
  return (
    typeof arg === 'function' &&
    arg[BEZIER_ICON_SYMBOL] === true
  )
}

export function createBezierIcon(source: IconSource): BezierIcon {
  source[BEZIER_ICON_SYMBOL] = true
  return source as BezierIcon
}
