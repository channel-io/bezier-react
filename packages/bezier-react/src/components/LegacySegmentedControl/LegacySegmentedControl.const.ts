import {
  Typography,
  type css,
} from '~/src/foundation'

import { LegacySegmentedControlSize } from './LegacySegmentedControl.types'

export const SIZE_TO_HEIGHT: Record<LegacySegmentedControlSize, number> = {
  [LegacySegmentedControlSize.XS]: 24,
  [LegacySegmentedControlSize.S]: 28,
  [LegacySegmentedControlSize.M]: 36,
  [LegacySegmentedControlSize.L]: 44,
}

export const SIZE_TO_PADDING: Record<LegacySegmentedControlSize, number> = {
  [LegacySegmentedControlSize.XS]: 1,
  [LegacySegmentedControlSize.S]: 2,
  [LegacySegmentedControlSize.M]: 2,
  [LegacySegmentedControlSize.L]: 4,
}

export const DIVIDER_WIDTH = 1
export const DIVIDER_SIDE_MARGIN = 6

export const SIZE_TO_DIVIDER_VERTICAL_MARGIN: Record<LegacySegmentedControlSize, number> = {
  [LegacySegmentedControlSize.XS]: 1,
  [LegacySegmentedControlSize.S]: 2,
  [LegacySegmentedControlSize.M]: 2,
  [LegacySegmentedControlSize.L]: 4,
}

export const SIZE_TO_OPTION_TYPOGRAPHY: Record<LegacySegmentedControlSize, ReturnType<typeof css>> = {
  [LegacySegmentedControlSize.XS]: Typography.Size13,
  [LegacySegmentedControlSize.S]: Typography.Size14,
  [LegacySegmentedControlSize.M]: Typography.Size14,
  [LegacySegmentedControlSize.L]: Typography.Size14,
}
