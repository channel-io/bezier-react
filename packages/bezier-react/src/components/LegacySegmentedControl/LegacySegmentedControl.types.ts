import type {
  BoxSizingUnit,
  ExplicitDefaulting,
} from '~/src/types/CSS'
import type {
  ActivatableProps,
  BezierComponentProps,
  ChildrenProps,
  OptionItemHostProps,
} from '~/src/types/ComponentProps'

import type { FormComponentProps } from '~/src/components/Forms/Form.types'

type SegmentedControlBaseProps = BezierComponentProps & ChildrenProps & FormComponentProps

type CSSSizingProperty = number | string | ExplicitDefaulting | BoxSizingUnit

export enum LegacySegmentedControlSize {
  XS = 'xs',
  S = 's',
  M = 'm',
  L = 'l',
}

interface LegacySegmentedControlOptions {
  width?: CSSSizingProperty

  /**
   * Size variant of this SegmentedControl.
   *
   * @default SegmentedControlSize.M
   */
  size?: LegacySegmentedControlSize
}

export default interface LegacySegmentedControlProps extends
  SegmentedControlBaseProps,
  OptionItemHostProps<number>,
  LegacySegmentedControlOptions {}

export interface LegacySegmentedControlItemProps extends
  SegmentedControlBaseProps,
  Pick<ActivatableProps, 'active'> {}
