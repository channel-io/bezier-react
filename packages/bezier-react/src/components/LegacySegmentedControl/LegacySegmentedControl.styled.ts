import {
  Transition,
  css,
  styled,
} from '~/src/foundation'

import disabledOpacity from '~/src/constants/DisabledOpacity'
import type { BezierComponentProps } from '~/src/types/ComponentProps'
import { toLength } from '~/src/utils/style'

import {
  SIZE_TO_DIVIDER_VERTICAL_MARGIN,
  SIZE_TO_HEIGHT,
  SIZE_TO_PADDING,
} from './LegacySegmentedControl.const'
import { LegacySegmentedControlSize } from './LegacySegmentedControl.types'
import type { LegacySegmentedControlItemProps } from './LegacySegmentedControl.types'

interface StyledWrapperProps extends BezierComponentProps {
  disabled?: boolean
  size: LegacySegmentedControlSize
  wrapperWidth: number | string
}

interface StyledOptionItemWrapperProps extends LegacySegmentedControlItemProps {
  size: LegacySegmentedControlSize
}

interface StyledIndicatorProps {
  size: LegacySegmentedControlSize
}

interface StyledDividerProps {
  size: LegacySegmentedControlSize
  hidden: boolean
}

const heightStyle = css<{ size: LegacySegmentedControlSize }>`
  height: ${({ size }) => SIZE_TO_HEIGHT[size]}px;
`

const indicatorHeightStyle = css<{ size: LegacySegmentedControlSize }>`
  height: ${({ size }) => SIZE_TO_HEIGHT[size] - (SIZE_TO_PADDING[size] * 2)}px;
`

const verticalPaddingStyle = css<{ size: LegacySegmentedControlSize }>`
  padding: ${({ size }) => SIZE_TO_PADDING[size]}px 0;
`

const wrapperRoundingStyle = css<{ size: LegacySegmentedControlSize }>`
  ${({ foundation, size }) => ({
    [LegacySegmentedControlSize.XS]: foundation?.rounding.round6,
    [LegacySegmentedControlSize.S]: foundation?.rounding.round8,
    [LegacySegmentedControlSize.M]: foundation?.rounding.round8,
    [LegacySegmentedControlSize.L]: foundation?.rounding.round12,
  })[size]}
`

const itemRoundingStyle = css<{ size: LegacySegmentedControlSize }>`
  ${({ foundation, size }) => ({
    [LegacySegmentedControlSize.XS]: foundation?.rounding.round4,
    [LegacySegmentedControlSize.S]: foundation?.rounding.round6,
    [LegacySegmentedControlSize.M]: foundation?.rounding.round6,
    [LegacySegmentedControlSize.L]: foundation?.rounding.round6,
  })[size]}
`

const indicatorRoundingStyle = css<{ size: LegacySegmentedControlSize }>`
  ${({ foundation, size }) => ({
    [LegacySegmentedControlSize.XS]: foundation?.rounding.round4,
    [LegacySegmentedControlSize.S]: foundation?.rounding.round6,
    [LegacySegmentedControlSize.M]: foundation?.rounding.round6,
    [LegacySegmentedControlSize.L]: foundation?.rounding.round8,
  })[size]}
`

export const Wrapper = styled.div<StyledWrapperProps>`
  position: relative;
  box-sizing: border-box;
  display: inline-flex;
  flex-direction: row;
  align-items: center;
  width: ${({ wrapperWidth }) => toLength(wrapperWidth, '100%')};
  min-width: 50px;
  overflow: hidden;
  background-color: ${props => props.foundation?.theme?.['bg-black-lighter']};
  ${({ disabled }) => disabled && `
    opacity: ${disabledOpacity};
  `}
  ${({ foundation }) => foundation?.transition?.getTransitionsCSS('background-color')};

  ${heightStyle}
  ${wrapperRoundingStyle}
  ${verticalPaddingStyle}
`

export const OptionItemWrapper = styled.div<StyledOptionItemWrapperProps>`
  position: absolute;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  overflow: hidden;
  color: ${props => (
    props.active
      ? props.foundation?.theme?.['txt-black-darkest']
      : props.foundation?.theme?.['txt-black-dark']
  )};
  cursor: ${props => {
    if (props.disabled) { return 'not-allowed' }
    if (props.active) { return 'auto' }
    return 'pointer'
  }};
  ${({ foundation }) => foundation?.transition?.getTransitionsCSS(['background-color', 'color'])};
  user-select: none;

  &:hover {
    ${props => ((!props.disabled && !props.active) ? `
      background-color: ${props.foundation?.theme?.['bg-black-lighter']};
    ` : '')}
  }

  ${heightStyle}
  ${itemRoundingStyle}
`

export const Indicator = styled.div<StyledIndicatorProps>`
  position: absolute;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  cursor: default;
  ${({ foundation }) => (
    foundation?.transition?.getTransitionsCSS('transform', Transition.TransitionDuration.M)
  )};
  will-change: transform;

  ${indicatorHeightStyle}
  ${indicatorRoundingStyle}
`

export const IndicatorBox = styled.div`
  ${({ foundation }) => foundation?.elevation?.ev1()};
  width: 100%;
  height: 100%;
  background-color: ${({ foundation }) => foundation?.theme?.['bg-white-high']};
`

export const DividerContainer = styled.div<StyledDividerProps>`
  position: absolute;
  top: ${({ size }) => SIZE_TO_DIVIDER_VERTICAL_MARGIN[size]}px;
  bottom: ${({ size }) => SIZE_TO_DIVIDER_VERTICAL_MARGIN[size]}px;

  ${({ hidden }) => hidden && `
    visibility: hidden;
  `}
`
