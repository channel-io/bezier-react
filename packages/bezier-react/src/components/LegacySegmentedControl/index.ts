import LegacySegmentedControl from './LegacySegmentedControl'
import { LegacySegmentedControlSize } from './LegacySegmentedControl.types'
import type LegacySegmentedControlProps from './LegacySegmentedControl.types'

export type {
  LegacySegmentedControlProps,
}

export {
  LegacySegmentedControl,
  LegacySegmentedControlSize,
}
