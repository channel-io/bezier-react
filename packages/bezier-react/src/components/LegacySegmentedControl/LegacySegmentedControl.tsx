import React, {
  type Ref,
  forwardRef,
  useCallback,
  useEffect,
  useMemo,
  useState,
} from 'react'

import { useResizeDetector } from 'react-resize-detector'
import { v4 as uuid } from 'uuid'

import useMergeRefs from '~/src/hooks/useMergeRefs'
import { noop } from '~/src/utils/function'
import { range } from '~/src/utils/number'
import { isNumber } from '~/src/utils/type'

import { Divider } from '~/src/components/Divider'
import useFormFieldProps from '~/src/components/Forms/useFormFieldProps'
import { Text } from '~/src/components/Text'

import {
  DIVIDER_SIDE_MARGIN,
  DIVIDER_WIDTH,
  SIZE_TO_OPTION_TYPOGRAPHY,
  SIZE_TO_PADDING,
} from './LegacySegmentedControl.const'
import type LegacySegmentedControlProps from './LegacySegmentedControl.types'
import { LegacySegmentedControlSize } from './LegacySegmentedControl.types'

import * as Styled from './LegacySegmentedControl.styled'

export const SEGMENTED_CONTROL_TEST_ID = 'bezier-react-segmented-control'
const SEGMENTED_CONTROL_OPTION_ITEM_TEST_ID_PREFIX = 'bezier-react-segmented-control-option-item'
export const segmentedControlOptionItemTestId = (index: number) =>
  [SEGMENTED_CONTROL_OPTION_ITEM_TEST_ID_PREFIX, index.toString()].join('-')

const itemWidth = (width: number, numItems: number, size: LegacySegmentedControlSize): number =>
  (width - (2 * SIZE_TO_PADDING[size]) - ((numItems - 1) * DIVIDER_WIDTH)) / Math.max(1, numItems)

const itemLeft = (width: number, numItems: number, size: LegacySegmentedControlSize) =>
  (index: number): number => (
    SIZE_TO_PADDING[size]
      + (itemWidth(width, numItems, size) * index)
      + (DIVIDER_WIDTH * index)
  )

const dividerLeft = (width: number, numItems: number, size: LegacySegmentedControlSize) =>
  (index: number): number => (
    SIZE_TO_PADDING[size]
      + (itemWidth(width, numItems, size) * index)
      + (DIVIDER_WIDTH * (index - 1))
      - DIVIDER_SIDE_MARGIN
  )

function LegacySegmentedControl(
  {
    testId = SEGMENTED_CONTROL_TEST_ID,
    width = '100%',
    size = LegacySegmentedControlSize.M,
    /* OptionItemHost props */
    selectedOptionIndex = 0,
    onChangeOption = noop,
    /* HTMLAttribute props */
    children,
    ...rest
  }: LegacySegmentedControlProps,
  forwardedRef: Ref<any>,
) {
  const {
    disabled,
    ...ownProps
  } = useFormFieldProps(rest)

  const [currentIndex, setCurrentIndex] = useState<number>(selectedOptionIndex)
  const [wrapperWidth, setWrapperWidth] = useState<number>(0)

  const numItems = useMemo(() => React.Children.count(children), [children])

  const optionItemWidth = itemWidth(wrapperWidth, numItems, size)
  const optionItemLeft = useMemo(() => (
    itemLeft(wrapperWidth, numItems, size)
  ), [
    wrapperWidth,
    numItems,
    size,
  ])
  const dividerItemLeft = useMemo(() => (
    dividerLeft(wrapperWidth, numItems, size)
  ), [
    wrapperWidth,
    numItems,
    size,
  ])

  const updateDimensions = useCallback((_width?: number) => {
    if (isNumber(_width)) { setWrapperWidth(_width) }
  }, [])

  const { ref: resizeDetectorRef } = useResizeDetector({ onResize: updateDimensions })

  const wrapperRef = useMergeRefs(resizeDetectorRef, forwardedRef)

  useEffect(() => {
    if (isNumber(selectedOptionIndex)) {
      setCurrentIndex(selectedOptionIndex)
    }
  }, [selectedOptionIndex])

  const handleClickOptionItem = useCallback((index: number) => {
    setCurrentIndex(index)
    onChangeOption(index)
  }, [onChangeOption])

  const renderOption = useCallback((Element: React.ReactNode, index: number) => (
    <Styled.OptionItemWrapper
      key={uuid()}
      data-testid={segmentedControlOptionItemTestId(index)}
      disabled={disabled}
      active={index === currentIndex}
      size={size}
      style={{
        width: optionItemWidth,
        left: optionItemLeft(index),
      }}
      onClick={() => (disabled ? noop : handleClickOptionItem(index))}
    >
      <Text
        typo={SIZE_TO_OPTION_TYPOGRAPHY[size]}
        bold
      >
        { Element }
      </Text>
    </Styled.OptionItemWrapper>
  ), [
    disabled,
    size,
    currentIndex,
    optionItemWidth,
    optionItemLeft,
    handleClickOptionItem,
  ])

  const Content = useMemo(() => (
    React.Children.map(children, renderOption)
  ), [
    children,
    renderOption,
  ])

  const Dividers = useMemo(() => (
    range(1, numItems)
      .map(index => (
        <Styled.DividerContainer
          key={`divider-${index.toString(36)}`}
          size={size}
          hidden={index === currentIndex || index === currentIndex + 1}
          style={{
            left: dividerItemLeft(index),
          }}
        >
          <Divider orientation="vertical" />
        </Styled.DividerContainer>
      ))
  ), [
    size,
    numItems,
    currentIndex,
    dividerItemLeft,
  ])

  const IndicatorComponent = useMemo(() => (
    <Styled.Indicator
      size={size}
      style={{
        width: optionItemWidth,
        transform: `translateX(${optionItemLeft(currentIndex)}px)`,
      }}
    >
      <Styled.IndicatorBox />
    </Styled.Indicator>
  ), [
    size,
    currentIndex,
    optionItemWidth,
    optionItemLeft,
  ])

  return (
    <Styled.Wrapper
      {...ownProps}
      ref={wrapperRef}
      disabled={disabled}
      size={size}
      wrapperWidth={width}
      data-testid={testId}
    >
      { IndicatorComponent }
      { Content }
      { Dividers }
    </Styled.Wrapper>
  )
}

export default forwardRef(LegacySegmentedControl)
