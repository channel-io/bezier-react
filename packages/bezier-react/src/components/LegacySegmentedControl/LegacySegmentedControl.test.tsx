import React from 'react'

import { LightFoundation } from '~/src/foundation'

import disabledOpacity from '~/src/constants/DisabledOpacity'
import { range } from '~/src/utils/number'
import { render } from '~/src/utils/test'

import { Text } from '~/src/components/Text'

import LegacySegmentedControl, {
  SEGMENTED_CONTROL_TEST_ID,
  segmentedControlOptionItemTestId,
} from './LegacySegmentedControl'
import type LegacySegmentedControlProps from './LegacySegmentedControl.types'
import ResizeObserver from './__mocks__/ResizeObserver'

describe('LegacySegmentedControl', () => {
  const defaultProps: Partial<LegacySegmentedControlProps> = {
    width: 144,
    selectedOptionIndex: 1,
    children: range(4).map((i) => <Text key={i}>{ i }</Text>),
  }

  const renderComponent = (props?: Partial<LegacySegmentedControlProps>) => render(
    <LegacySegmentedControl {...defaultProps} {...props} />,
  )

  let oldWindowResizeObserver

  beforeAll(() => {
    oldWindowResizeObserver = window.ResizeObserver
    window.ResizeObserver = ResizeObserver
  })

  afterAll(() => {
    window.ResizeObserver = oldWindowResizeObserver
  })

  it('Snapshot', () => {
    const { getByTestId } = renderComponent()
    const rendered = getByTestId(SEGMENTED_CONTROL_TEST_ID)

    expect(rendered).toMatchSnapshot()
  })

  it('has active and non-active items with correct style', () => {
    const { getByTestId } = renderComponent({ selectedOptionIndex: 1 })

    const firstItem = getByTestId(segmentedControlOptionItemTestId(0))
    const secondItem = getByTestId(segmentedControlOptionItemTestId(1))

    expect(firstItem).toHaveStyle(`color: ${LightFoundation.theme['txt-black-dark']};`)
    expect(secondItem).toHaveStyle(`color: ${LightFoundation.theme['txt-black-darkest']};`)
  })

  it('should have correct disabled style', () => {
    const { getByTestId } = renderComponent({ disabled: true })
    const rendered = getByTestId(SEGMENTED_CONTROL_TEST_ID)

    expect(rendered).toHaveStyle(`opacity: ${disabledOpacity};`)
  })

  it('has active and non-active items with correct style, even if disabled', () => {
    const { getByTestId } = renderComponent({ disabled: true, selectedOptionIndex: 1 })

    const firstItem = getByTestId(segmentedControlOptionItemTestId(0))
    const secondItem = getByTestId(segmentedControlOptionItemTestId(1))

    expect(firstItem).toHaveStyle(`color: ${LightFoundation.theme['txt-black-dark']};`)
    expect(secondItem).toHaveStyle(`color: ${LightFoundation.theme['txt-black-darkest']};`)
  })
})
