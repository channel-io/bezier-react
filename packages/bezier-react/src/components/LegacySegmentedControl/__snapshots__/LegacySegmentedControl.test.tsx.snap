// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegacySegmentedControl Snapshot 1`] = `
.c4 {
  font-size: 1.4rem;
  line-height: 1.8rem;
  margin: 0px 0px 0px 0px;
  font-style: normal;
  font-weight: bold;
  color: inherit;
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
  -webkit-transition-timing-function: cubic-bezier(.3,0,0,1);
  transition-timing-function: cubic-bezier(.3,0,0,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: color;
  transition-property: color;
}

.c5 {
  font-size: 1.5rem;
  line-height: 2rem;
  -webkit-letter-spacing: -0.1px;
  -moz-letter-spacing: -0.1px;
  -ms-letter-spacing: -0.1px;
  letter-spacing: -0.1px;
  margin: 0px 0px 0px 0px;
  font-style: normal;
  font-weight: normal;
  color: inherit;
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
  -webkit-transition-timing-function: cubic-bezier(.3,0,0,1);
  transition-timing-function: cubic-bezier(.3,0,0,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: color;
  transition-property: color;
}

.c8 {
  overflow: hidden;
  border-radius: 1px;
  background-color: #00000014;
  width: 1px;
  height: calc(100% - 12px);
  margin: 6px;
}

.c0 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 144px;
  min-width: 50px;
  overflow: hidden;
  background-color: #0000000D;
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
  -webkit-transition-timing-function: cubic-bezier(.3,0,0,1);
  transition-timing-function: cubic-bezier(.3,0,0,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 36px;
  overflow: hidden;
  border-radius: 8px;
  padding: 2px 0;
}

.c3 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  color: #00000066;
  cursor: pointer;
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
  -webkit-transition-timing-function: cubic-bezier(.3,0,0,1);
  transition-timing-function: cubic-bezier(.3,0,0,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,color;
  transition-property: background-color,color;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 36px;
  overflow: hidden;
  border-radius: 6px;
}

.c3:hover {
  background-color: #0000000D;
}

.c6 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  color: #000000D9;
  cursor: auto;
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
  -webkit-transition-timing-function: cubic-bezier(.3,0,0,1);
  transition-timing-function: cubic-bezier(.3,0,0,1);
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
  -webkit-transition-property: background-color,color;
  transition-property: background-color,color;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 36px;
  overflow: hidden;
  border-radius: 6px;
}

.c1 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: default;
  -webkit-transition-delay: 0ms;
  transition-delay: 0ms;
  -webkit-transition-timing-function: cubic-bezier(.3,0,0,1);
  transition-timing-function: cubic-bezier(.3,0,0,1);
  -webkit-transition-duration: 300ms;
  transition-duration: 300ms;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  will-change: transform;
  height: 32px;
  overflow: hidden;
  border-radius: 6px;
}

.c2 {
  background-color: #FFFFFF;
  box-shadow: inset 0 0 2px 0 #FFFFFF1F, 0 0 2px 1px #0000000D, 0 1px 2px #00000014;
  width: 100%;
  height: 100%;
  background-color: #FFFFFF;
}

.c7 {
  position: absolute;
  top: 2px;
  bottom: 2px;
  visibility: hidden;
}

.c9 {
  position: absolute;
  top: 2px;
  bottom: 2px;
}

<div
  class="c0"
  data-testid="bezier-react-segmented-control"
>
  <div
    class="c1"
    style="width: -1.75px; transform: translateX(1.25px);"
  >
    <div
      class="c2"
    />
  </div>
  <div
    class="c3"
    data-testid="bezier-react-segmented-control-option-item-0"
    style="width: -1.75px; left: 2px;"
  >
    <span
      class="c4"
      data-testid="bezier-react-text"
    >
      <span
        class="c5"
        data-testid="bezier-react-text"
      >
        0
      </span>
    </span>
  </div>
  <div
    class="c6"
    data-testid="bezier-react-segmented-control-option-item-1"
    style="width: -1.75px; left: 1.25px;"
  >
    <span
      class="c4"
      data-testid="bezier-react-text"
    >
      <span
        class="c5"
        data-testid="bezier-react-text"
      >
        1
      </span>
    </span>
  </div>
  <div
    class="c3"
    data-testid="bezier-react-segmented-control-option-item-2"
    style="width: -1.75px; left: 0.5px;"
  >
    <span
      class="c4"
      data-testid="bezier-react-text"
    >
      <span
        class="c5"
        data-testid="bezier-react-text"
      >
        2
      </span>
    </span>
  </div>
  <div
    class="c3"
    data-testid="bezier-react-segmented-control-option-item-3"
    style="width: -1.75px; left: -0.25px;"
  >
    <span
      class="c4"
      data-testid="bezier-react-text"
    >
      <span
        class="c5"
        data-testid="bezier-react-text"
      >
        3
      </span>
    </span>
  </div>
  <div
    class="c7"
    hidden=""
    style="left: -5.75px;"
  >
    <div
      aria-orientation="vertical"
      class="c8"
      data-orientation="vertical"
      data-testid="bezier-react-divider"
      orientation="vertical"
      role="separator"
    />
  </div>
  <div
    class="c7"
    hidden=""
    style="left: -6.5px;"
  >
    <div
      aria-orientation="vertical"
      class="c8"
      data-orientation="vertical"
      data-testid="bezier-react-divider"
      orientation="vertical"
      role="separator"
    />
  </div>
  <div
    class="c9"
    style="left: -7.25px;"
  >
    <div
      aria-orientation="vertical"
      class="c8"
      data-orientation="vertical"
      data-testid="bezier-react-divider"
      orientation="vertical"
      role="separator"
    />
  </div>
</div>
`;
